doc/source/release-notes/Percona-XtraDB-Cluster-5.7.20-29.24.rst

```diff
@@ -0,0 +1,121 @@
+
+.. note:: Due to new package dependency,
+   Ubuntu/Debian users should use ``apt-get dist-upgrade``, ``apt upgrade``,
+   or ``apt-get install percona-xtradb-cluster-57`` to upgrade.
+
+Percona XtraDB Cluster 5.7.20-29.24 is now the current release,
+based on the following:
+
+* `Percona Server 5.7.20-18 <https://www.percona.com/doc/percona-server/5.7/release-notes/Percona-Server-5.7.20-18.html>`_
+
+* Galera Replication library 3.22
+
+* Galera/Codership WSREP API Release 5.7.20
+
+All Percona software is open-source and free.
+
+NEW FEATURES:
+=============
+
+* Ubuntu 17.10 Artful Aardvark is now supported.
+
+* Freezing gcache purge was implemented to facilitate node
+  joining through IST, avoiding time consuming SST process.
+
+* :jirabug:`PXC-822`: an usability improvement was made in timeout error
+  messages making them to name the configuration variable which caused this
+  timeout.
+
+* :jirabug:`PXC-868`: on the Joiner, during SST, "tmpdir" variable under
+  [sst] section can be used to specify temporary SST files storage different
+  from the default datadir/.sst one.
+
+* A new variable :variable:`wsrep_last_applied` in addition
+  to :variable:`wsrep_last_committed` one was introduced to clearly separate
+  last committed and last applied transaction numbers.
+
+
+Fixed Bugs
+==========
+
+* :jirabug:`PXC-889`: fixed an issue where a node with an invalid value for
+  :variable:`wsrep_provider` was allowed to start up and operate in standalone
+  mode, which could lead to data inconsistency. The node will now abort in
+  this case. Bug fixed :bug:`1728774`
+
+* :jirabug:`PXC-806`: fixed early read of the query_id, ensuring valid
+  id to be assigned to the running thread.
+
+* :jirabug:`PXC-850`: ensured that a node, because of data inconsistency,
+  isolates itself before leaving the cluster, thus allowing pending nodes
+  to re-evaluate the quorum. Bug fixed :bug:`1704404`
+
+* :jirabug:`PXC-867`: ``wsrep_sst_rsync`` script was overwriting wsrep debug
+  configuration setting making it not to be taken into account.
+
+* Fixed formatting issue in the error message appearing
+  when SST is not possible due to a timeout. Bug fixed :bug:`1720094`
+
+* :jirabug:`PXC-874`: PXC acting as async slave reported unhandled transaction
+  errors, namely "Rolling back unfinished transaction".
+
+* :jirabug:`PXC-875`: fixed an issue where toggling :variable:`wsrep_provider`
+  off and on failed to reset some internal variables and resulted in PXC
+  logging an "Unsupported protocol downgrade" warning. Bug fixed :bug:`1379204`
+
+* :jirabug:`PXC-877`: fixed PXC hang caused by an internal deadlock.
+
+* Thread failed to mark exit from the InnoDB server concurrency and therefore
+  never got un-register in InnoDB concurrency system.
+
+* :jirabug:`PXC-879`: wrong count took place when ``LOAD DATA`` commands used
+  with GTIDs were executed on one node making other nodes to receive less rows
+  than the first one. Bug fixed :bug:`1741818`
```

Comment:
Fixed a bug where a ``LOAD DATA`` command used with GTIDS was executed on one node, but the other nodes would receive less rows than the first one.